csharp/extractor/Semmle.Extraction.CSharp/Entities/Field.cs

```diff
@@ -85,7 +85,18 @@ public override void Populate(TextWriter trapFile)
                 Where(n => n.EqualsValue != null))
             {
                 // Mark fields that have explicit initializers.
-                var expr = new Expression(new ExpressionInfo(Context, Type, Context.Create(initializer.EqualsValue.Value.FixedLocation()), Kinds.ExprKind.FIELD_ACCESS, this, child++, false, null));
+                var constValue = Context
+                    .GetModel(initializer.EqualsValue.Value)
+                    .GetConstantValue(initializer.EqualsValue.Value);
+
+                var constValueToStore = constValue.HasValue
+                    ? Expression.ValueAsString(constValue.Value)
+                    : null;
+
+                var expr = new Expression(new ExpressionInfo(
```

Comment:
Why aren't we extracting the expression itself as a subexpression? 
```
enum MyEnum
{
  A = 1+2+3,
  B
}
```

If I read this correctly, then `1+2+3` is extracted here as a field access instead of an addition.

Reply:
You are right, we could extract that.

Reply:
It probably doesn't matter too much. One case that might be interesting if the constant expression contains a reference to a constant and in some check we need all the references to that constant.